### ISSUE TITLE
Enabling debugging causes XHR Poll Error

### ISSUE THREAD

**Issue by gboysko** [socketio/socket.io-client#1056]
I have a socket.io Server which I am testing in an integration test. I am encountering the odd behavior of a client-side error ("xhr poll error"), only when I enable the `socket.io:socket` and `socket.io:client` via the `DEBUG` environment variable. Without the debug enabled, the tests runs fine.

### You want to:

* [x] report a *bug*
* [ ] request a *feature*

### Current behaviour

When I run my automated integration test with `DEBUG=socket.io:socket,socket.io:client`, then the test will fail with an `xhr poll error` midway (2-3 seconds into the test) through the extended test. 

### Steps to reproduce (if the current behaviour is a bug)

This will be hard for me to provide a reproducible test case for you as it involves our full server and automated unit test suite.

### Expected behaviour

The client connection does not fail regardless of the `DEBUG` settings.

### Setup
- OS: Mac OS
- browser: None--the client is running inside a Jasmine/Jest environment.
- socket.io version: Client is 1.5.1.

### Other information (e.g. stacktraces, related issues, suggestions how to fix)

Nothing else to provide.

**Comment by darrachequesne** [comment]
Do you get any exception, either client-side or server-side? Without a proper way to reproduce, you'll understand that it will be hard to fix.

Note: which version of `debug` are you using? The one that comes with `socket.io`? There has been some issues lately, some releases (`2.4.x`) that were not working at all: https://github.com/visionmedia/debug/releases

**Comment by gboysko** [comment]
Thanks, @darrachequesne. I understand the difficulty in diagnosing this problem given the lack of details. I don't see any exceptions (either client or server-side), which seems unusual. I don't fully understand the explanation of the "xhr poll error", other the poll request failed for an unspecified reason. I have not specified the transports to use in my automated test case and therefore, I guess it defaults to polling first. I will try to use the `websocket` transport to see if that has an effect. 

FWIW, I am using `debug` version 2.3.3, the "official" one that ships independent of socket.io. 

I will update the version of `debug` as well and report back.

**Comment by gboysko** [comment]
Clarification: The debugging that I enable (which causes the test to hang) is **server-side** DEBUG modules, not *client-side*. Still investigating the various client-side modules for additional insight. I thought that `engine.io-client:polling-xhr` would tell me something, but no luck. I probably don't understand the implementation as well as I should.

**Comment by gboysko** [comment]
Hmmm... This looks promising. By enabling `engine.io-client:socket`, I see this right before the failure:

```
Tue, 10 Jan 2017 23:55:19 GMT engine.io-client:socket socket error {"type":"TransportError","description":503}
Tue, 10 Jan 2017 23:55:19 GMT engine.io-client:socket socket close with reason: "transport error"
Tue, 10 Jan 2017 23:55:19 GMT engine.io-client:socket probe transport "websocket" failed because of error: socket closed
```

Does this mean that the Server failed?

**Comment by gboysko** [comment]
One other update: my client was running 1.5.1, but my server was 1.7.1, so I upgraded my client to 1.7.1 (actually 1.7.2 was selected by NPM), but the error remains.

**Comment by darrachequesne** [comment]
@gboysko http 503 means the polling request has failed with `Service unavailable` reason. It seems the request doesn't actually reach the server, are there any kind of proxy in between?

Nonetheless the link with `debug` package remains quite subtle.

**Comment by gboysko** [comment]
@darrachequesne Thanks for insight. There is no proxy. In fact, the test is composed of several Jasmine unit tests which interact with the same Socket.io Server instance. This particular test involves the exchange of hundreds of messages. It always fails in the same place.

It "feels" like the Server is busy sending a lot of messages and somehow the client doesn't get a response fast enough. The time period since the last message received was on the order of 15ms or so. So it is not like the Server hasn't responding in seconds. 

As I write this, I am wondering if the problem is more likely with the client or server. Clearly, when running, the only thing that changes is the server-side DEBUG env var (adding `socket.io:socket` and `socket.io:client`). With logging on, clearly, the Server is generated more output, but would be enough to cause a `Service Unavailable` error. There must be options which control how long the client will wait on a polling request. I'll try to find these and continue to experiment with other settings.

Finally, I will definitely upgrade the `debug` module on the Server side to the latest to rule that out.

**Comment by gboysko** [comment]
**Update:** With either `socket.io:socket` **or** `socket.io:client` enabled, the problem does not occur. It only consistently occurs when both modules are specified.

I updated the Server's `debug` module to the latest, but it had no change whatsoever to the problem. I am ruling that out.

I see that `engine.io-client` has a `requestTimeout` option (which defaults to `0`?), but I don't see a way to pass that through the socket.io Manager class. 

I tried to specify the `websocket` transport for my socket.io client (via `transports: ['websocket', 'polling']`), but that seems to fail immediately. Not sure if a Jasmine-based unit test can support a Web Sockets connection to the Server).
